### DISCUSSION TITLE
AvaloniaUI and AndroidTV remote controller

### DISCUSSION THREAD

**Discussion by michalss** [AvaloniaUI/Avalonia#9483] [Q&A] [answered]
HI,

I would like to know if there is any support for keyboard/remote controller/game controller navigation in Avalonia ? I would like to build applications for AndroidTV as well but, unfortunately they most of the time require to use remote controller or game controller.

Regards

**Comment by maxkatz6** [comment, marked as answer]
No, it's not currently supported. 
You would need to handle Android events and manually raise raw keyboard events, and possibly replace IKeyboardDevice/FocusManager implementation to work with XY navigation. Which isn't trivial.
Related tickets: https://github.com/AvaloniaUI/Avalonia/issues/6945 https://github.com/AvaloniaUI/Avalonia/issues/7607

**Reply by michalss** [reply]
thx for info. But it will be done in some point right ? If so i can only hope it will be soon.. :) thx anyway

**Reply by timunie** [reply]
Hi @michalss I remember that a guy on telegram shared some proof of concept, but it's not yet a stable API. Maybe you can ask there if he can share his results with you, but I can't promise.  

If you need this feature in a short or medium term, you have two options:
1. Get yourself the source of Avalonia and implement it on your own. If that is good enough, a PR is welcome, but I can't guarantee that this PR will be merged. 
2. You can always request a feature to the team. If possible at all, you will get an offer which you can order or not.

**Reply by michalss** [reply]
thank you ill try telegram. Still think, core dev team should focus in such a feature, since this platform is growing very fast and there would be more and more ppl who would be pushed to deploy applications to AndroidTV .
